app/controllers/user/subscriptions_controller.rb

```diff
@@ -50,8 +50,7 @@ def create
             sign_in_path: new_user_sessions_path(host: current_site.domain)
           )
         end
-
-        redirect_to request.referrer
+        redirect_to(request.referrer || root_path)
```

Comment:
There's a new way to do this!

https://blog.bigbinary.com/2016/02/29/rails-5-improves-redirect_to_back-with-redirect-back.html